### ISSUE TITLE
What is the function signature for the variable text

### ISSUE THREAD

**Issue by lo-co** [Templarian/ui.bootstrap.contextMenu#31]
Just a question, but I don't see labels showing up here.  You have the following code where you expect a string for the text in the first argument but account for a function if the type is not a ``string`` (line 25):

```javascript
 var text = typeof item[0] == 'string' ? item[0] : item[0].call($scope, $scope, event, model);
```

What is the intention of the function for ``item[0]`` (I assume to return a string) and what is this signature?

You have a similar definition below this (line 30):

```javascript
var enabled = angular.isDefined(item[2]) ? item[2].call($scope, $scope, event, text, model) : true;
```

It seems to have the same signature as the function call above.  What is the intention of this line?

**Comment by Templarian** [comment]
https://github.com/Templarian/ui.bootstrap.contextMenu#menu-options The first code example shows the signature.

Basically it's an array that is overloaded to allow either a `string` or `function` as some people need dynamically named items depending on what they are right clicking.

```javascript
function ($itemScope, $event, model) {
    return $itemScope.item.name;
}
```

Note: as shown in the example `model` is an optional attribute if you need to easily pass in an additional object.

**Comment by Templarian** [comment]
Sorry forgot to answer the second part of the question. The 2nd line of code is for enable/disabling of the context menu items. Very useful when right clicking items from a list where some have different abilities.

**Comment by lo-co** [comment]
OK.  Thanks Austin.  Now for a question born out of my ignorance - why does the example function signature not resemble the called function signature?

**Comment by Templarian** [comment]
So, in JavaScript the first parameter is actually the function scope or `this`.

```javascript
function foo(p1) {
 // this.bar == 'hello world!'
 // p1 == 'cool right'
}
foo.call({bar: 'hello world!'}, 'cool right')
```

**Comment by Templarian** [comment]
Oh wait, I think you found a bug... that 2nd param should be...

`item[2].call($scope, $scope, event, model, text)` Hmm... that needs to be fixed.

**Comment by lo-co** [comment]
Perfect...thanks.

**Comment by Templarian** [comment]
Okay, I've fixed this in `v0.9.5`, but I'm not going to document it since I'm not sure exactly the purpose of passing in the text would be. It also gets confusing since `model` is optional.

**Comment by Templarian** [comment]
`v0.9.6` is out now also. Closing this issue.
